wlroots/ffi_build.py

```diff
@@ -780,6 +780,62 @@ def has_xwayland() -> bool:
 uint32_t wlr_keyboard_get_modifiers(struct wlr_keyboard *keyboard);
 """
 
+# types/wlr_presentation_time.h
+CDEF += """
+struct wlr_surface;
+
+struct wlr_output;
+struct wlr_output_event_present;
+
+struct wlr_presentation {
```

Comment:
If you put `typedef int32_t clockid_t;` above this line, it should work. We do already have this typedef below for `types/wlr_seat.h` below, we might want to just move it up to the top and just start out ```CDEF = """\
typedef int32_t clockid_t;
"""

# backend.h
CDEF += """
```

Reply:
> If you put `typedef int32_t clockid_t;` above this line, it should work. We do already have this typedef below for `types/wlr_seat.h` below, we might want to just move it up to the top and just start out
> 
> ```
> CDEF = """\
> typedef int32_t clockid_t;
> """
> 
> # backend.h
> CDEF += """
> ```

Alright I'll get this done asap